PurchasesCoreSwift/Attribution/AttributionTypeFactory.swift

```diff
@@ -0,0 +1,123 @@
+//
+//  AttributionTypeFactory.swift
+//  PurchasesCoreSwift
+//
+//  Created by Juanpe Catalán on 9/7/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+
+// TODO(Post-migration): switch this back to internal the class and all these protocols and properties.
+
+public typealias AttributionDetailsBlock = ([String: Any]?, Error?) -> Void
+
+@objc public enum FakeTrackingManagerAuthorizationStatus: Int {
+    case notDetermined = 0
+    case restricted
+    case denied
+    case authorized
+}
+
+class FakeAdClient: NSObject {
+    // We need this method to be available as an optional implicitly unwrapped method for `AnyClass`.
+    @objc static func sharedClient() -> FakeAdClient {
+        FakeAdClient()
+    }
+
+    // We need this method to be available as an optional implicitly unwrapped method for `AnyClass`.
+    @objc func requestAttributionDetails(_ completionHandler: AttributionDetailsBlock) {
+        // do nothing
```

Comment:
Would it make sense to add some logging here, like a warning that this method was called and it shouldn't have been?

Reply:
Well, this method would execute if the original class couldn't be instantiated. If so, the factory does an early return and log a message indicating that the framework hasn't been imported yet. 

But, TBH, I think if we add a warning message, it's not too much, and It could give us more context for weird scenarios 👍🏼